src/seedu/addressbook/parser/Parser.java

```diff
@@ -174,6 +178,44 @@ private Command prepareDelete(String args) {
         }
     }
 
+    private Command prepareUpdate(String args)  {
+        try{
+            String[] data = args.split(" ");
+            String index = data[1];
+            String rest = "";
+            for(int i = 2; i < data.length; i++)
+            {
+                rest = rest + data[i] + " ";
+            }
+            final int targetIndex = parseArgsAsDisplayedIndex(index);
+            final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(rest.trim());
+            if (!matcher.matches()) {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
```

Comment:
Instead of using AddCommand's MESSAGE_USAGE, maybe try adding a separate "error message" to be more specific in the error handling process so when the system prints out the "message" you will be able to identify which exact part of your code it is running (instead of having the same message appear for 2 different situations).